pkg/azclient/auth.go

```diff
@@ -155,17 +229,21 @@ func NewAuthProvider(armConfig *ARMClientConfig, config *AzureAuthConfig, client
 
 	// UserAssignedIdentityCredentials authentication
 	if computeCredential == nil && len(config.AADMSIDataPlaneIdentityPath) > 0 {
-		computeCredential, err = dataplane.NewUserAssignedIdentityCredential(context.Background(), config.AADMSIDataPlaneIdentityPath, dataplane.WithClientOpts(azcore.ClientOptions{Cloud: clientOption.Cloud}))
+		computeCredential, err = dataplane.NewUserAssignedIdentityCredential(
+			context.Background(),
+			config.AADMSIDataPlaneIdentityPath,
+			dataplane.WithClientOpts(azcore.ClientOptions{Cloud: clientOption.Cloud}),
+		)
 		if err != nil {
 			return nil, err
 		}
 	}
 
 	return &AuthProvider{
-		ComputeCredential:     computeCredential,
-		NetworkCredential:     networkTokenCredential,
-		MultiTenantCredential: multiTenantCredential,
-		CloudConfig:           clientOption.Cloud,
+		ComputeCredential:              computeCredential,
+		AdditionalComputeClientOptions: additionalComputeClientOptions,
```

Comment:
Is there any reference to this field?

Reply:
Will be used in L418 in https://github.com/zarvd/cloud-provider-azure/blob/b24b7a4a34c24199494a18b1859db89584558594/pkg/provider/azure.go#L401-L427. 

Compared to the existing one https://github.com/kubernetes-sigs/cloud-provider-azure/blob/a2e9ea1d7ff931f1f25346d7883f5b92939b9019/pkg/provider/azure.go#L405-L453

Reply:
Could you please extend auth policy in runtime just like how arm policy did?https://github.com/Azure/azure-sdk-for-go/blob/d453e307d3ab003538976638a366b5a918aaa164/sdk/azcore/arm/runtime/policy_bearer_token.go

Reply:
It provide check auth method and the behavior is validated.

Reply:
wait a sec, I had queued a E2E test to run for it.

Reply:
The test failed using the arm policy. Will keep it this way as this impl for cross-tenant had been tested in AKS.